### PR TITLE
Basic python extension types for image, point2d, and box2d

### PR DESCRIPTION
**Pull request by changhiskhan**
This depends on #100 . The unit tests were segfaulting until I rebuilt liblance using #100 branch.

### CODE REVIEW

File: python/lance/tests/test_types.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import platform
+import pytest
+
+import lance
+import pyarrow as pa
+
+from lance.types import *
+
+
+@pytest.mark.skipif(
```
Comment (eddyxu) [inline python/lance/tests/test_types.py:10]: we can skip the entire file

https://stackoverflow.com/a/42512169

File: python/lance/tests/test_types.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import platform
```
Comment (eddyxu) [inline python/lance/tests/test_types.py:1]: License and `isort`

File: python/lance/types.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+import pandas as pd
+
+import pyarrow as pa
+
+
+__all__ = ["ImageType", "ImageUriType", "ImageBinaryType", "Point2dType", "Box2dType"]
+
+
+# Arrow extension type
```
Comment (eddyxu) [inline python/lance/types.py:29]: move imports above `__all__ = [...]` ?

Seems need to run isort as well.

File: python/lance/types.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+"""
+from abc import ABC, abstractproperty
+import platform
+
+import pandas as pd
+
+import pyarrow as pa
+
+
+__all__ = ["ImageType", "ImageUriType", "ImageBinaryType", "Point2dType", "Box2dType"]
```
Comment (eddyxu) [inline python/lance/types.py:26]: Should we not expose ` ImageType`, which is the base class?

File: python/lance/types.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        super(LanceType, self).__init__(storage_type, extension_name)
+
+
+class ImageType(LanceType):
+    @abstractproperty
+    def storage_type(self):
+        pass
+
+    def __arrow_ext_serialize__(self):
+        return b""
```
Comment (eddyxu) [inline python/lance/types.py:48]: Curious what does this function do?

Comment (changhiskhan) [reply python/lance/types.py:48]: For parametrized types this returns extra information (from the examples in Arrow docs)

File: python/lance/types.py (hunk 4 above)
Comment (changhiskhan) [reply python/lance/types.py:26]: ImageType.from_storage allows the user to get the correct subtype directly from storage array so they don't need to think about whether they need a uri or binary type

**Review by eddyxu — APPROVED** [review]

**Pull request merged**